src/parse_util.rs

```diff
@@ -1120,6 +1120,17 @@ pub fn parse_util_detect_errors_in_ast(
             if jcc.conjunction.has_source() && jcc.job.try_source_range().is_none() {
                 has_unclosed_conjunction = true;
             }
+            let continuations = jcc
+                .parent()
+                .and_then(|job_conj_list| job_conj_list.as_job_conjunction_continuation_list());
+            if let Some(continuations) = continuations {
+                for continuation in continuations.contents() {
+                    if continuation.job.bg.is_some() {
```

Comment:
looking at only continuations that have `&` is not fully correct because it means we fail to
check for `foo & && bar`. The first command is not a continuation.
Consider these test cases:

```diff
diff --git a/src/parse_util.rs b/src/parse_util.rs
index 2ab2d1001..3d3e11909 100644
--- a/src/parse_util.rs
+++ b/src/parse_util.rs
@@ -1822,7 +1822,7 @@ pub fn parse_util_expand_variable_error(
 }

 /// Error message for use of backgrounded commands before and/or.
-const BOOL_AFTER_BACKGROUND_ERROR_MSG: &str =
+pub(crate) const BOOL_AFTER_BACKGROUND_ERROR_MSG: &str =
     "The '%ls' command can not be used immediately after a backgrounded job";

 /// Error message for backgrounded commands as conditionals.
diff --git a/src/tests/parse_util.rs b/src/tests/parse_util.rs
index 3b6748f5a..d4b7cd14f 100644
--- a/src/tests/parse_util.rs
+++ b/src/tests/parse_util.rs
@@ -5,7 +5,7 @@ use crate::parse_constants::{
     ERROR_NOT_ARGV_AT, ERROR_NOT_ARGV_COUNT, ERROR_NOT_ARGV_STAR, ERROR_NOT_PID, ERROR_NOT_STATUS,
     ERROR_NO_VAR_NAME,
 };
-use crate::parse_util::parse_util_detect_errors;
+use crate::parse_util::{parse_util_detect_errors, BOOL_AFTER_BACKGROUND_ERROR_MSG};
 use crate::tests::prelude::*;
 use crate::wchar::prelude::*;
 use crate::wchar_ext::WExt;
@@ -72,4 +72,9 @@ fn test_error_messages() {
     validate!("echo foo\"$\"bar", ERROR_NO_VAR_NAME);
     validate!("echo \"foo\"$\"bar\"", ERROR_NO_VAR_NAME);
     validate!("echo foo $ bar", ERROR_NO_VAR_NAME);
+    validate!("echo 1 & && echo 2", BOOL_AFTER_BACKGROUND_ERROR_MSG);
+    validate!(
+        "echo 1 && echo 2 & && echo 3",
+        BOOL_AFTER_BACKGROUND_ERROR_MSG
+    );
 }
```

Reply:
Yep, just noticed the error. It should be fine now.